### PR TITLE
fix(addon/a11y): run scan only when addon tab is active

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
Issue: [#5973 A11y addon freezes the UI](https://github.com/storybooks/storybook/issues/5973)

## What I did
- On a11y addon initialization, there is no listener set on story rendered event to avoid running the scan when the tab is inactive.
- Hide report when a scan is running instead of displaying outdated report.

The resulting behavior : 
- on init, index listen to REQUEST event **only** to perform the a11y scan
- A11y panel listen to RESULT event to update the displayed report
- A11y panel listen to STORY_RENDERED event to emit a REQUEST only if the tab is active

Bonus : Add unit tests on a11y panel component.

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes
- Does this need a new example in the kitchen sink apps? No 
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Review by ndelangen — DISMISSED** [review]

File: addons/a11y/src/index.ts [hunk 1]
```diff
@@ -56,7 +55,6 @@ export const withA11y: StoryWrapper = (getStory, context) => {
   return getStory(context);
 };
 
-channel.on(STORY_RENDERED, () => run(setup.element, setup.config, setup.options));
```
Comment (ndelangen) [inline addons/a11y/src/index.ts]: Shouldn't we clear the page when the story changed?

The result might be incorrect..

Comment (jsomsanith) [reply addons/a11y/src/index.ts]: We still have an old state in the A11YPanel component.
But 
- if the tab is inactive, the old report is not displayed
- when tab becomes active, the old report is not rendered, I changed the behavior to display only the status button when it is running (on tab activation, a REQUEST event is sent, and status is running)
- if tab is active, on story change, a REQUEST event is sent, hiding the old report (see previous point) while running

File: addons/a11y/src/components/A11YPanel.tsx [hunk 2]
```diff
@@ -142,29 +142,31 @@ export class A11YPanel extends Component<A11YPanelProps, A11YPanelState> {
...
-              {
-                label: <Incomplete>{incomplete.length} Incomplete</Incomplete>,
-                panel: (
-                  <Report passes={false} items={incomplete} empty="No a11y incomplete found." />
-                ),
-              },
-            ]}
-          />
-        </ScrollArea>
+        {status === 'running' ? null : (
```
Comment (shilman) [inline addons/a11y/src/components/A11YPanel.tsx]: Nit-picking but maybe we can show a loading spinner here?

**Review by shilman — APPROVED** [review]
LGTM! Love that you added tests .. bravo 👏

File: addons/a11y/src/components/A11YPanel.tsx (hunk 2 above)
Comment (jsomsanith) [reply addons/a11y/src/components/A11YPanel.tsx]: Done :)

**Comment by shilman** [comment]
There are too many changes to the a11y addon, so this will only be available in `5.1.x`

**Pull request merged**